### ISSUE TITLE
[tsql] Parsing errors... WHEN multiple CREATE PROCEDURE statements stored within the same file.

### ISSUE THREAD

**Issue by dmoore247** [sqlfluff/sqlfluff#4943]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I can run sqlfluff lint on both files individually, each file containing just one CREATE PROCEDURE call; 
However most of our client code base have multiple create procedure calls per file.
When this happens we get a `WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?` and that file fails the lint-ing.

### Expected Behaviour

Should just pass the parsing best case; else make it lint style. Please do something other than fail on the parse.

### Observed Behaviour

```
==== sqlfluff ====
sqlfluff:                2.1.1 python:                 3.11.4
implementation:        cpython verbosity:                   1
dialect:                  tsql templater:               jinja
rules:                                all
==== readout ====

== [tsql-four.sql] FAIL                                                                      
L:  13 | P:   1 |  PRS | Line 13, Position 1: Found unparsable section: 'CREATE              
                       | PROC foo @CustomerName NVARCHAR(5...'                               
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
== [tsql-one.sql] PASS                                                                       
== [tsql-three.sql] PASS                                                                     
== [tsql-two.sql] PASS                                                                       
==== summary ====                                                                            
violations:        1 status:         FAIL                                                    
All Finished 📜 🎉!  
```

### How to reproduce

Command:
`sqlfluff lint --verbose --dialect tsql tsql*.sql`

tsql-four.sql:
```sql
CREATE PROC raw_abdc_finance.sapecc_dataintegrity_t001 (
    @StagingTableName [varchar](255),
    @RawTableName [varchar](255),
    @MetaDataText [varchar](MAX)
)
AS
BEGIN
    EXEC ingestdataintorawfromstagingtable
        @StagingTableName, @RawTableName, @MetaDataText
END;
-- passes sql fluff

CREATE PROC foo @CustomerName NVARCHAR(50) AS
BEGIN
    SELECT
        a,
        b,
        c
    FROM dev
END;
```

### Dialect

tsql

### Version

2.1.1

### Configuration

default / no config overrides.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by kajarit** [comment]
The two create statements must be separated with a GO keyword, see:
_Note: GO is not a Transact-SQL statement_
https://learn.microsoft.com/en-us/sql/t-sql/language-elements/sql-server-utilities-statements-go?view=sql-server-ver16

**Comment by peterbud** [comment]
With the latest `sqlfluff` the above file with multiple `CREATE PROC` are properly parsed (even without `GO` statament).
